Parsers/ASimAuthentication/Parsers/ASimAuthenticationSentinelOne.yaml

```diff
@@ -0,0 +1,182 @@
+Description: |
+  This ASIM parser supports normalizing SentinelOne logs to the ASIM Authentication normalized schema. SentinelOne events are captured through SentinelOne data connector which ingests SentinelOne server objects such as Threats, Agents, Applications, Activities, Policies, Groups, and more events into Microsoft Sentinel through the REST API.
+ParserName: ASimAuthenticationSentinelOne
+EquivalentBuiltInParser: _ASim_Authentication_SentinelOne
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventResultDetailsLookup = datatable (comments_s: string, EventResultDetails: string)
+  [
+      "invalid 2FA code", "Incorrect password",
+      "IP/User mismatch", "No such user or password",
+      "invalid password", "Incorrect password",
+      "user temporarily locked 2FA attempt", "User locked",
+      "no active site", "Other"
+  ];
+  let EventFieldsLookup = datatable (
+      activityType_d: real,
+      EventType: string,
+      EventResult: string,
+      EventOriginalType: string
+  )
+  [
+      27, "Logon", "Success", "User Logged In",
+      33, "Logoff", "Success", "User Logged Out",
+      133, "Logon", "Failure", "Existing User Login Failure",
+      134, "Logon", "Failure", "Unknown User Login",
+      139, "Logon", "Failure", "User Failed to Start an Unrestricted Session",
+      3629, "Logon", "Success", "Login Using Saved 2FA Recovery Code"
+  ];
+  let EventTypeLookup = datatable (alertInfo_eventType_s: string, EventType: string)
+      [
+      "WINLOGONATTEMPT", "Logon",
+      "WINLOGOFFATTEMPT", "Logoff"
+  ];
+  let EventSubTypeLookup = datatable (alertInfo_loginType_s: string, EventSubType: string)
+  [
+      "BATCH","System",
+      "CACHED_INTERACTIVE", "Interactive",
+      "CACHED_REMOTE_INTERACTIVE", "RemoteInteractive",
+      "CACHED_UNLOCK", "System",
+      "INTERACTIVE", "Interactive",
+      "NETWORK_CLEAR_TEXT", "Remote",
+      "NETWORK_CREDENTIALS", "Remote",
+      "NETWORK", "Remote",
+      "REMOTE_INTERACTIVE", "RemoteInteractive",
+      "SERVICE", "Service",
+      "SYSTEM", "System",
+      "UNLOCK", "System"
+  ];
+  let DeviceTypeLookup = datatable (agentDetectionInfo_machineType_s: string, TargetDeviceType: string)
+  [
+      "desktop", "Computer",
+      "server", "Computer",
+      "laptop", "Computer",
+      "kubernetes node", "Other",
+      "unknown", "Other"
+  ];
+  let parser = (disabled: bool=false) {
+      let alldata = SentinelOne_CL
+          | where not(disabled);
+      let activitydata = alldata
+          | where event_name_s == "Activities."
+              and activityType_d in (27, 33, 133, 134, 139, 3629)
+          | parse-kv DataFields_s as (ipAddress: string, username: string, userScope: string, accountName: string, fullScopeDetails: string, fullScopeDetailsPath: string, role: string, scopeLevel: string, source: string, sourceType: string) with (pair_delimiter=",", kv_delimiter=":", quote='"')
```

Comment:
'role' could potentially be TargetUserType?
https://learn.microsoft.com/en-us/azure/sentinel/normalization-about-schemas#additional-user-fields

Reply:
As discussed we have mapped the role directly with TargetUserType when the value is similar to the allowed values, kept empty for role="null" and mapped to "Other" for other type of values.